owncloudApp/src/androidTest/java/com/owncloud/android/settings/camerauploads/OCSettingsLocalFolderPickerTest.kt

```diff
@@ -72,4 +80,19 @@ class OCSettingsLocalFolderPickerTest {
         ).parent.split("/").last()
         onView(withText(path)).check(matches(isDisplayed()))
     }
+
+    fun waitForLocalPickerActivity(){
+        var textVisible = false
+        while (textVisible == false) {
+            try {
+                onView(withContentDescription(R.string.abc_action_bar_up_description)).check(matches(isDisplayed()))
+                onView(withId(R.id.folder_picker_btn_home)).check(matches(isDisplayed()))
+                onView(withId(R.id.folder_picker_btn_choose)).check(matches(isDisplayed()))
+                onView(withId(R.id.folder_picker_btn_cancel)).check(matches(isDisplayed()))
+                textVisible = true
+            } catch (e: NoMatchingViewException) {
```

Comment:
This is nasty, I hope I find time to show you a proposal how you can solve this specific task. (Not all tests !)

Reply:
I'm checking also better solutions. thanks.